mlflow/spark.py

```diff
@@ -235,21 +240,29 @@ def log_model(
     try:
         spark_model.save(posixpath.join(model_dir, _SPARK_MODEL_PATH_SUB))
     except Py4JError:
-        return Model.log(
-            artifact_path=artifact_path,
-            flavor=mlflow.spark,
-            spark_model=spark_model,
-            conda_env=conda_env,
-            code_paths=code_paths,
-            dfs_tmpdir=dfs_tmpdir,
-            sample_input=sample_input,
-            registered_model_name=registered_model_name,
-            signature=signature,
-            input_example=input_example,
-            await_registration_for=await_registration_for,
-            pip_requirements=pip_requirements,
-            extra_pip_requirements=extra_pip_requirements,
-        )
+        artifact_repo = get_artifact_repository(artifact_uri=run_root_artifact_uri)
+        if (
+            isinstance(artifact_repo, DatabricksArtifactRepository)
+            and databricks_utils.is_mlflowdbfs_available()
+        ):
+            # TODO: test we're saving model metadata when mlflowdbfs is available.
```

Comment:
Let's make sure to add this test coverage by mocking the presence / behavior of `mlflowdbfs`.